src/pybind/mgr/debug/module.py

```diff
@@ -0,0 +1,64 @@
+from mgr_module import CLIReadCommand, HandleCommandResult, MgrModule, Option
+from threading import Event
+from typing import cast, Any, Optional, TYPE_CHECKING
+import errno
+import epdb
+
+def bp(condition: callable = None):
+    if condition:
+        epdb.set_trace_cond(condition)
+    else:
+        epdb.set_trace()
+
+class DebugModule(MgrModule):
+
+    # These are "native" Ceph options that this module cares about.
+    NATIVE_OPTIONS = [
+        'mgr_tick_period',
+    ]
+
+    on = False
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+
+        # set up some members to enable the serve() method and shutdown()
+        self.run = True
+        self.event = Event()
+
+        # ensure config options members are initialized; see config_notify()
+        self.config_notify()
+        self.port = 7777
```

Comment:
instead of a hardcoded value I think it would be better to have this port as configurable variable.

Reply:
This one is a dev tool so I don't see a point of adding auth.

Reply:
ok, but then we need to ensure that this module won't run in production environments... By default everything beneath pybind/mgr will be installed (no allowlist/blocklist for supported mgr modules). Maybe something to explore here.